src/SourceBuild/content/repo-projects/source-build-reference-packages.proj

```diff
@@ -25,4 +25,24 @@
   </Target>
 
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+
+  <UsingTask AssemblyFile="$(LeakDetectionTasksAssembly)" TaskName="MarkAndCatalogPackages" />
+
+  <Target Name="PoisonSbrp"
+          Condition="'$(EnablePoison)' == 'true'"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(RepoCompletedSemaphorePath)PoisonCurrentSbrp.complete"
+          AfterTargets="Build">
+    <ItemGroup>
+      <CurrentSbrp Include="$(ReferencePackagesDir)**/*.nupkg" />
```

Comment:
After poisoning `vmr/prereqs/packages/reference` at [this commit](a4e1c155baee463805c5af89adb4cb1165df9ad0) (from the preview7 branch before changes related to https://github.com/dotnet/source-build/issues/3599 were merged), I've discovered that the following reference assemblies get marked as being poisoned:

```
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/Microsoft.Extensions.DependencyInjection.Abstractions.dll
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/Microsoft.Extensions.DependencyInjection.dll
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/Microsoft.Extensions.FileSystemGlobbing.dll
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/Microsoft.Extensions.Logging.Abstractions.dll
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/Microsoft.Extensions.Logging.dll
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/Microsoft.Extensions.Options.dll
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/Microsoft.Extensions.Primitives.dll
sdk/8.0.100-preview.7.23376.1/DotnetTools/dotnet-format/System.IO.Pipelines.dll
```

In comparison, the following reference assemblies do not appear in the poison-usage report but continue to appear in the produced sdk. I'm still working to determine where these reference assemblies might be being pulled from so that they can be poisoned and marked:

```
sdk/8.0.100-preview.7.23376.1/Microsoft.Win32.SystemEvents.dll
sdk/8.0.100-preview.7.23376.1/System.ComponentModel.Composition.dll
sdk/8.0.100-preview.7.23376.1/System.Drawing.Common.dll
sdk/8.0.100-preview.7.23376.1/System.Security.Permissions.dll
sdk/8.0.100-preview.7.23376.1/System.Windows.Extensions.dll
sdk/8.0.100-preview.7.23376.1/FSharp/Microsoft.Win32.SystemEvents.dll
sdk/8.0.100-preview.7.23376.1/FSharp/System.CodeDom.dll
sdk/8.0.100-preview.7.23376.1/FSharp/System.Drawing.Common.dll
sdk/8.0.100-preview.7.23376.1/FSharp/System.Resources.Extensions.dll
sdk/8.0.100-preview.7.23376.1/FSharp/System.Security.Cryptography.Xml.dll
sdk/8.0.100-preview.7.23376.1/FSharp/System.Security.Permissions.dll
sdk/8.0.100-preview.7.23376.1/FSharp/System.Windows.Extensions.dll
sdk/8.0.100-preview.7.23376.1/Containers/tasks/net8.0/Microsoft.Win32.SystemEvents.dll
sdk/8.0.100-preview.7.23376.1/Containers/tasks/net8.0/System.Drawing.Common.dll
sdk/8.0.100-preview.7.23376.1/Containers/tasks/net8.0/System.Security.Permissions.dll
sdk/8.0.100-preview.7.23376.1/Containers/tasks/net8.0/System.Windows.Extensions.dll
```

Reply:
Regarding the dotnet-format reference assemblies, I believe this PR will address most if not all of these - https://github.com/dotnet/format/pull/1930